CMakeLists.txt

```diff
@@ -497,5 +494,11 @@ include_directories(.)
 # generates an installable makefile.
 install(FILES velox/type/Type.h DESTINATION "include/velox")
 
+# Adding this down here prevents warnings in dependencies from stopping the
+# build
+if("${TREAT_WARNINGS_AS_ERRORS}")
```

Comment:
I was under impression that the order shouldnt matter since cmake runs out of band - so only after the cmake process are the builds undertaken (maybe because we use macros with resolve_dependency its in process ?)

Reply:
The order matters as the compile commands are constructed during configure time and (generally) in the sequence they appear in the cmake file (that's why we add boost before folly as folly depends on boost so we want to configure/create the boost targets first). Concurrency also plays a role and the actual build order is determined at the end of configure time but at that point the compile commands for the TUs have already been generated.

Adding `-Werror` just before the velox subdirs are run ensures that we don't get hung up on errors produced by warnings in dependencies and don't have to manually suppress warnings we are not responsible for.